testsrc/org/mozilla/javascript/tests/Issue1297FunctionNameTest.java

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.Scriptable;
+
+/**
+ * Test that we can redefine a function's name.
+ */
+public class Issue1297FunctionNameTest {
+	private static final String source = "'use strict';" +
+			"function X() {};\n" +
+			"Object.defineProperty(X, 'name', {value: 'y', configurable: true, writable: true});" +
+			"X.name";
+
+	@Test
+	public void canSetFunctionNameInCompiledClasses() {
+		try (Context cx = Context.enter()) {
+			Scriptable scope = cx.initStandardObjects(null);
+			Object result = cx.evaluateString(scope,
+					source,
+					"test", 1, null);
+			assertEquals("y", result);
+		}
+	}
+
+	@Test
+	public void canSetFunctionNameInInterpreter() {
+		try (Context cx = Context.enter()) {
```

Comment:
please use Utils.runWithAllOptimizationLevels like many of the other tests are doing

Reply:
And please fix formating, see https://github.com/mozilla/rhino#code-formatting

Reply:
Done and done. 

Totally missed the code formatting, my bad. :-)